src/modules/cartPage.ts

```diff
@@ -0,0 +1,134 @@
+'use strict';
+import { OnButtonClick, OnProductClick } from './interfaces/customTypes';
+import { IProduct } from './interfaces/product.interface';
+
+type OnChangeAmount = (event: Event, id: number) => void;
+
+export class CartPage {
+    private parent: Element;
+    private cartDataLayout: string;
+    private productsData: IProduct[];
+    private selectedProducts: number[];
+    private onChangeAmount: OnChangeAmount;
+    private onProductClick: OnProductClick;
+    private onButtonClick: OnButtonClick;
```

Comment:
Here and everywhere else: consider `readonly` for underlying immutability.  Approach is the same you use for `let/var` and `const`.

Reply:
I'll fix all cases next PR.